### ISSUE TITLE
`cell_color` parameter in `gene_trends` does not work as expected

### ISSUE THREAD

**Issue by Marius1311** [theislab/cellrank#486]
In the API, is says: " Color of the cells when not visualizing absorption probabilities. Only used when same_plot=True". So I'm setting `same_plot=True` and I call:
```python
model = cr.ul.models.GAMR(adata_raw, n_knots=7)

fig_kwargs = {'model': model,
              'data_key': 'magic_imputed_data', 
              'genes': ['Pdx1'], 
              'same_plot': True,
             'cell_color': 'clusters',
              'time_key': 'palantir_pseudotime', 
              'weight_threshold': (0.05, 0.05)
             }

cr.pl.gene_trends(adata_raw, **fig_kwargs)
```
however, this does not show me the actual cells:
![image](https://user-images.githubusercontent.com/24494650/107046137-9a02ea80-67c6-11eb-916c-d01b2fadb071.png)

So I'm trying to get around this by explicitly passing `hide_cells=False`:
```python
model = cr.ul.models.GAMR(adata_raw, n_knots=7)

fig_kwargs = {'model': model,
              'data_key': 'magic_imputed_data', 
              'genes': ['Pdx1'], 
              'same_plot': True,
              'hide_cells': False,
              'cell_color': 'clusters',
              'time_key': 'palantir_pseudotime', 
              'weight_threshold': (0.05, 0.05)
             }

cr.pl.gene_trends(adata_raw, **fig_kwargs)
```
Doesn't make a difference: 
![image](https://user-images.githubusercontent.com/24494650/107046361-d3d3f100-67c6-11eb-9731-d46dd1d730e4.png)

Let's remove the `cell_color` argument from the function call:
```python
model = cr.ul.models.GAMR(adata_raw, n_knots=7)

fig_kwargs = {'model': model,
              'data_key': 'magic_imputed_data', 
              'genes': ['Pdx1'], 
              'same_plot': True,
              'hide_cells': False,
              # 'cell_color': 'clusters',
              'time_key': 'palantir_pseudotime', 
              'weight_threshold': (0.05, 0.05)
             }

cr.pl.gene_trends(adata_raw, **fig_kwargs)
```
![image](https://user-images.githubusercontent.com/24494650/107046436-e8b08480-67c6-11eb-900c-d3b1e3f0d9c0.png)

There are the cells, but now just shown in black. 

#### Versions:
cellrank==1.2.0 scanpy==1.6.0 anndata==0.7.4 numpy==1.19.5 numba==0.52.0 scipy==1.5.3 pandas==1.1.3 scikit-learn==0.23.2 statsmodels==0.12.0 python-igraph==0.8.3 scvelo==0.2.2 pygam==0.8.0 matplotlib==3.2.2 seaborn==0.11.0
> ...

**Comment by michalk8** [comment]
`cell_color` was never intendet to pull data from `adata`, it was just to change the color of the cells from the default `black`.
Not sure why matplotlib silently fails, but that's why you don't see any colors (try `cell_color='red'`, and it will work).
However, I see that this can be quite useful, will include this (for data from `.obs` and `.var`), but I will also retain the old behavior not to break stuff.

**Comment by Marius1311** [comment]
thanks! I don't really see any point if being able to have all cells in one color like red, but if it's easier for now because of testing and stuff then let's keep it and let's just add the extra option to color in `.obs` and `.var_names` attributes (`.var` won't work because it has the wrong dimensions - #genes, not #cells)

**Comment by Marius1311** [comment]
thanks for looking into this @michalk8!

**Comment by michalk8** [comment]
closed via #487
